components/webdriver_server/lib.rs

```diff
@@ -347,6 +410,131 @@ impl<'de> Visitor<'de> for TupleVecMapVisitor {
     }
 }
 
+fn compute_tick_duration(tick_actions: &ActionSequence) -> u64 {
+    let mut duration = 0;
+    match &tick_actions.actions {
+        ActionsType::Null { actions } => {
+            for action in actions.iter() {
+                let NullActionItem::General(GeneralAction::Pause(pause_action)) = action;
+                duration = cmp::max(duration, pause_action.duration.unwrap_or(0));
+            }
+        },
+        ActionsType::Pointer {
+            parameters: _,
+            actions,
+        } => {
+            for action in actions.iter() {
+                if let PointerActionItem::General(GeneralAction::Pause(pause_action)) = action {
+                    duration = cmp::max(duration, pause_action.duration.unwrap_or(0));
+                } else if let PointerActionItem::Pointer(PointerAction::Move(move_action)) = action
+                {
+                    duration = cmp::max(duration, move_action.duration.unwrap_or(0));
+                }
+            }
+        },
+        _ => (),
+    }
+    duration
+}
+
+// https://w3c.github.io/webdriver/#dfn-dispatch-tick-actions
+fn dispatch_tick_actions(
+    session: &mut WebDriverSession,
+    tick_actions: &ActionSequence,
+    tick_duration: u64,
+) -> WebDriverResult<WebDriverResponse> {
+    // Step 1.1
+    let source_id = tick_actions.id.as_ref().unwrap();
+    // Step 1.2 - 1.6
+    match &tick_actions.actions {
+        ActionsType::Null { actions } => {
+            for action in actions.iter() {
+                // Step 1.3 - 1.4
+                session
+                    .input_state_table
+                    .entry(source_id.to_string())
+                    .or_insert(InputSourceState::Null);
```

Comment:
Let's add a `dispatch_general_action` helper which matches on the passed GeneralAction value, and we can call it from each of the duplicated pieces of code below this.

Reply:
I'm not sure if we can have a generic `dispatch_general_action` since the actions that need to be dispatched all have different types: https://searchfox.org/mozilla-central/source/testing/webdriver/src/actions.rs.

Reply:
All of the General variants contain an instance of GeneralAction. I'm proposing that dispatch_general_action accept a GeneralAction argument.

Reply:
Yes, that makes sense, I misunderstood the term `general` here.